src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs

```diff
@@ -268,12 +268,7 @@ public ViewViewComponentResult View<TModel>(TModel model)
         /// <returns>A <see cref="ViewViewComponentResult"/>.</returns>
         public ViewViewComponentResult View<TModel>(string viewName, TModel model)
         {
-            if (model != null)
-            {
-                ViewData.Model = model;
-            }
-
-            var viewData = new ViewDataDictionary<TModel>(ViewData);
+            var viewData = new ViewDataDictionary<TModel>(ViewData, model);
```

Comment:
Side note: This isn't quite the same as the previous code because the `ViewData.Model` value won't be updated in the current scope (for the view component). I.e. the view component would previously have seen a new `ViewData.Model` value if they had code after the `View()` call (same as similar code in a controller). Won't hit errors in the assignment either way because the `ViewDataDictionary` is always `ViewDataDictionary<object>`. See the [`ViewComponentContext` constructor](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/ViewComponentContext.cs#L81).

Doubt this is worth worrying about at all but wanted to call it out in case ~~someone~~ ~~@Eilon~~ someone disagrees.